packages/vite/src/node/plugins/resolve.ts

```diff
@@ -545,90 +545,101 @@ export function resolvePackageEntry(
   if (cached) {
     return cached
   }
-  let entryPoint: string | undefined | void
+  try {
```

Comment:
**suggestion:** This is a huge try-catch block. This leads to _if anything fails, do catch_.
Maybe we want to only surround the part that should really be catched.

Reply:
You mentioned `resolveExports` throws the error. Maybe then it's sufficient to only surround this part

```ts
// resolve exports field with highest priority
// using https://github.com/lukeed/resolve.exports
if (data.exports) {
  entryPoint = resolveExports(data, '.', options, targetWeb)
}
```

Reply:
I think that it would be appropriate to catch all errors. If an error happened elsewhere in this method I'd still want it to be handled the same way